Content/Items/Misc/Weapons.Soilgun.cs

```diff
@@ -0,0 +1,1488 @@
+            if (!Main.LocalPlayer.HasItem(ModContent.ItemType<Soilgun>()))
+                return;
+
+            if (ValidSoils.Contains(item.type))
+            {
+                TooltipLine tooltip = new TooltipLine(Mod, "SoilgunAmmoTooltip", "This item can be used as ammo for the Soilgun");
+                tooltips.Add(tooltip);
+                tooltip.OverrideColor = new Color(114, 81, 56);
+                if (item.type == Mod.Find<ModItem>("VitricSandItem").Type)
+                {
+                    TooltipLine infoTooltip = new TooltipLine(Mod, "AmmoInfoTooltip", "When used with the Soilgun, it will fire out volatile blocks of sand that stick to enemies and explode");
+                    tooltips.Add(infoTooltip);
+                    infoTooltip.OverrideColor = new Color(114, 81, 56);
+                    return;
+                }
+                switch (item.type)
+                {
+                    case ItemID.SandBlock: infoTooltip2 = new TooltipLine(Mod, "AmmoInfoTooltip", "When used with the Soilgun, it will fire out blocks of sand that split into many grains of sand upon death"); break;
+                    case ItemID.CrimsandBlock: infoTooltip2 = new TooltipLine(Mod, "AmmoInfoTooltip", "When used with the Soilgun, it will fire out blocks of Crimsand that steal life from hit enemies"); break;
+                    case ItemID.EbonsandBlock: infoTooltip2 = new TooltipLine(Mod, "AmmoInfoTooltip", "When used with the Soilgun it will fire out blocks of Ebonsand that apply stacks of Haunted to enemies\nHaunted does 1 DPS per haunted stack, maxing at 20\nWhen an enemy has 20 Haunted stacks, they will cause haunted apparitions to exorcise from them, getting rid of all their Haunted stacks"); break;
+                    case ItemID.PearlsandBlock: infoTooltip2 = new TooltipLine(Mod, "AmmoInfoTooltip", "When used with the Soilgun, it will fire out blocks of Pearlsand that home in on enemies"); break;
+                    case ItemID.DirtBlock: infoTooltip2 = new TooltipLine(Mod, "AmmoInfoTooltip", "When used with the Soilgun, it will fire out blocks of dirt"); break;
+                    case ItemID.SiltBlock: infoTooltip2 = new TooltipLine(Mod, "AmmoInfoTooltip", "When used with the Soilgun, it will fire out blocks of silt, that spawn coins upon hitting enemies"); break;
+                    case ItemID.SlushBlock: infoTooltip2 = new TooltipLine(Mod, "AmmoInfoTooltip", "When used with the Soilgun, it will fire out blocks of slush that cause hit enemies to have icicles impale them\nHitting and enemy with more than 15 icicles causes all icicles to shatter, causing large amounts of damage"); break;
+                }
+                tooltips.Add(infoTooltip2);
+                infoTooltip2.OverrideColor = new Color(114, 81, 56);
+            }
+        }
+    }
+
+    class SoilgunGlobalNPC : GlobalNPC
```

Comment:
This class will likely be depricated once backend support for stackable DoT is finalized and implemented. For now this is alright but ideally we want to reduce as many of these types of classes as possible.